src/NHibernate/Impl/SessionFactoryObjectFactory.cs

```diff
@@ -21,8 +22,8 @@ public static class SessionFactoryObjectFactory
 	{
 		private static readonly INHibernateLogger log;
 
-		private static readonly IDictionary<string, ISessionFactory> Instances = new Dictionary<string, ISessionFactory>();
-		private static readonly IDictionary<string, ISessionFactory> NamedInstances = new Dictionary<string, ISessionFactory>();
+		private static readonly IDictionary<string, ISessionFactory> Instances = new ConcurrentDictionary<string, ISessionFactory>();
+		private static readonly IDictionary<string, ISessionFactory> NamedInstances = new ConcurrentDictionary<string, ISessionFactory>();
```

Comment:
This looks to me legit, while all other changes look superfluous to me. May you isolate this in a standalone PR?

Reply:
Well, actually no, these changes are superfluous too, because all methods of the type are synchronized, meaning no concurrent access can occur.